### PR TITLE
Assert(valid) when ext_proc filter apply header mutations

### PR DESCRIPTION
**Pull request by yanjunxiang-google**
When ext_proc server sends back some header mutation settings which contain NULL characters, like /n, /r or /0, this will cause ext_proc filter ASSERT(valid) being fired up when Envoy is built in debug mode. 
In release mode, the ASSERT is no-op, however, we should add an explicit check to avoid such kind of headers get injected into Envoy.

Assert(valid) when ext_proc filter apply header mutations.

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by yanjunxiang-google** [comment]
This is a fuzzer issue:



Accepting input from '[STDIN]'
--
  | Usage for fuzzing: honggfuzz -P [flags] -- /mnt/scratch0/clusterfuzz/bot/builds/clusterfuzz-builds-honggfuzz_envoy_13d8ff3fd8b6e12ff5bbd32d951c40c9e1c6513f/revisions/ext_proc_unit_test_fuzz
  | [2023-05-22 14:48:58.839][2428][critical][assert] [./envoy/http/header_map.h:71] assert failure: valid().
  | AddressSanitizer:DEADLYSIGNAL
  | =================================================================
  | ==2428==ERROR: AddressSanitizer: ABRT on unknown address 0x05390000097c (pc 0x7a908660600b bp 0x7ffdaf9e1c90 sp 0x7ffdaf9e19b0 T0)
  | SCARINESS: 10 (signal)
  | #0 0x7a908660600b in raise /build/glibc-SzIz7B/glibc-2.31/sysdeps/unix/sysv/linux/raise.c:51:1
  | #1 0x7a90865e5858 in abort /build/glibc-SzIz7B/glibc-2.31/stdlib/abort.c:79:7
  | #2 0x1557b8f in Envoy::Http::LowerCaseString::LowerCaseString(std::__1::basic_string_view<char, std::__1::char_traits<char> >) envoy/http/header_map.h:71:5
  | #3 0x1685664 in Envoy::Extensions::HttpFilters::ExternalProcessing::MutationUtils::applyHeaderMutations(envoy::service::ext_proc::v3::HeaderMutation const&, Envoy::Http::HeaderMap&, bool, Envoy::Extensions::Filters::Common::MutationRules::Checker const&, Envoy::Stats::Counter&) /proc/self/cwd/source/extensions/filters/http/ext_proc/mutation_utils.cc:43:27
  | #4 0x165dc14 in Envoy::Extensions::HttpFilters::ExternalProcessing::ProcessorState::handleHeadersResponse(envoy::service::ext_proc::v3::HeadersResponse const&) /proc/self/cwd/source/extensions/filters/http/ext_proc/processor_state.cc:87:31
  | #5 0x163ae9e in Envoy::Extensions::HttpFilters::ExternalProcessing::Filter::onReceiveMessage(std::__1::unique_ptr<envoy::service::ext_proc::v3::ProcessingResponse, std::__1::default_delete<envoy::service::ext_proc::v3::ProcessingResponse> >&&) /proc/self/cwd/source/extensions/filters/http/ext_proc/ext_proc.cc:609:41
  | #6 0x14c2fbf in operator() /proc/self/cwd/test/extensions/filters/http/ext_proc/unit_test_fuzz/ext_proc_unit_test_fuzz.cc:86:31
  | #7 0x14c2fbf in __invoke<(lambda at test/extensions/filters/http/ext_proc/unit_test_fuzz/ext_proc_unit_test_fuzz.cc:82:28) &, envoy::service::ext_proc::v3::ProcessingRequest, bool> /usr/local/include/c++/v1/type_traits:3592:23
  | #8 0x14c2fbf in __call<(lambda at test/extensions/filters/http/ext_proc/unit_test_fuzz/ext_proc_unit_test_fuzz.cc:82:28) &, envoy::service::ext_proc::v3::ProcessingRequest, bool> /usr/local/include/c++/v1/__functional/invoke.h:61:9
  | #9 0x14c2fbf in operator() /usr/local/include/c++/v1/__functional/function.h:181:16
  | #10 0x14c2fbf in std::__1::__function::__func<Envoy::Extensions::HttpFilters::ExtProc::UnitTestFuzz::TestOneProtoInput(envoy::extensions::filters::http::ext_proc::unit_test_fuzz::ExtProcUnitTestCase const&)::$_0::operator()(Envoy::Extensions::HttpFilters::ExternalProcessing::ExternalProcessorCallbacks&, envoy::config::core::v3::GrpcService const&, Envoy::StreamInfo::StreamInfo const&) const::'lambda'(envoy::service::ext_proc::v3::ProcessingRequest&&, bool), std::__1::allocator<Envoy::Extensions::HttpFilters::ExtProc::UnitTestFuzz::TestOneProtoInput(envoy::extensions::filters::http::ext_proc::unit_test_fuzz::ExtProcUnitTestCase const&)::$_0::operator()(Envoy::Extensions::HttpFilters::ExternalProcessing::ExternalProcessorCallbacks&, envoy::config::core::v3::GrpcService const&, Envoy::StreamInfo::StreamInfo const&) const::'lambda'(envoy::service::ext_proc::v3::ProcessingRequest&&, bool)>, void (envoy::service::ext_proc::v3::ProcessingRequest&&, bool)>::operator()(envoy::service::ext_proc::v3::ProcessingRequest&&, bool&&) /usr/local/include/c++/v1/__functional/function.h:355:12
  | #11 0x1586861 in operator() /usr/local/include/c++/v1/__functional/function.h:508:16
  | #12 0x1586861 in operator() /usr/local/include/c++/v1/__functional/function.h:1185:12
  | #13 0x1586861 in ApplyImpl<const std::__1::function<void (envoy::service::ext_proc::v3::ProcessingRequest &&, bool)> &, std::__1::tuple<envoy::service::ext_proc::v3::ProcessingRequest &&, bool>, 0UL, 1UL> /proc/self/cwd/external/com_google_googletest/googlemock/include/gmock/internal/gmock-internal-utils.h:420:10
  | #14 0x1586861 in Apply<const std::__1::function<void (envoy::service::ext_proc::v3::ProcessingRequest &&, bool)> &, std::__1::tuple<envoy::service::ext_proc::v3::ProcessingRequest &&, bool> > /proc/self/cwd/external/com_google_googletest/googlemock/include/gmock/internal/gmock-internal-utils.h:429:10
  | #15 0x1586861 in Perform /proc/self/cwd/external/com_google_googletest/googlemock/include/gmock/gmock-actions.h:497:12
  | #16 0x1586861 in PerformAction<void (envoy::service::ext_proc::v3::ProcessingRequest &&, bool)> /proc/self/cwd/external/com_google_googletest/googlemock/include/gmock/gmock-spec-builders.h:1441:12
  | #17 0x1586861 in testing::internal::FunctionMocker<void (envoy::service::ext_proc::v3::ProcessingRequest&&, bool)>::UntypedPerformAction(void const*, void*) const /proc/self/cwd/external/com_google_googletest/googlemock/include/gmock/gmock-spec-builders.h:1556:12
  | #18 0x50c8339 in testing::internal::UntypedFunctionMockerBase::UntypedInvokeWith(void*) /proc/self/cwd/external/com_google_googletest/googlemock/src/gmock-spec-builders.cc:452:24
  | #19 0x1590021 in testing::internal::FunctionMocker<void (envoy::service::ext_proc::v3::ProcessingRequest&&, bool)>::Invoke(envoy::service::ext_proc::v3::ProcessingRequest&&, bool) /proc/self/cwd/external/com_google_googletest/googlemock/include/gmock/gmock-spec-builders.h:1593:15
  | #20 0x1585db3 in Envoy::Extensions::HttpFilters::ExtProc::UnitTestFuzz::MockStream::send(envoy::service::ext_proc::v3::ProcessingRequest&&, bool) /proc/self/cwd/test/extensions/filters/http/ext_proc/unit_test_fuzz/mocks.h:20:3
  | #21 0x162023f in Envoy::Extensions::HttpFilters::ExternalProcessing::Filter::onHeaders(Envoy::Extensions::HttpFilters::ExternalProcessing::ProcessorState&, Envoy::Http::RequestOrResponseHeaderMap&, bool) /proc/self/cwd/source/extensions/filters/http/ext_proc/ext_proc.cc:177:12
  | #22 0x16225a6 in Envoy::Extensions::HttpFilters::ExternalProcessing::Filter::decodeHeaders(Envoy::Http::RequestHeaderMap&, bool) /proc/self/cwd/source/extensions/filters/http/ext_proc/ext_proc.cc:195:23
  | #23 0x15c247b in Envoy::Http::FilterHeadersStatus Envoy::Extensions::HttpFilters::HttpFilterFuzzer::sendHeaders<Envoy::Http::StreamDecoderFilter>(Envoy::Http::StreamDecoderFilter*, test::fuzz::HttpData const&, bool) /proc/self/cwd/test/extensions/filters/http/common/fuzz/http_filter_fuzzer.h:145:46
  | #24 0x14d648f in void Envoy::Extensions::HttpFilters::HttpFilterFuzzer::runData<Envoy::Http::StreamDecoderFilter>(Envoy::Http::StreamDecoderFilter*, test::fuzz::HttpData const&) /proc/self/cwd/test/extensions/filters/http/common/fuzz/http_filter_fuzzer.h:102:31
  | #25 0x14bf07a in TestOneProtoInput /proc/self/cwd/test/extensions/filters/http/ext_proc/unit_test_fuzz/ext_proc_unit_test_fuzz.cc:93:10
  | #26 0x14bf07a in LLVMFuzzerTestOneInput /proc/self/cwd/test/extensions/filters/http/ext_proc/unit_test_fuzz/ext_proc_unit_test_fuzz.cc:38:1
  | #27 0x519eedb in main
  | #28 0x7a90865e7082 in __libc_start_main /build/glibc-SzIz7B/glibc-2.31/csu/libc-start.c:308:16
  | #29 0x140142d in _start
  |  
  | AddressSanitizer can not provide additional info.
  | SUMMARY: AddressSanitizer: ABRT (/lib/x86_64-linux-gnu/libc.so.6+0x4300b) (BuildId: 1878e6b475720c7c51969e69ab2d276fae6d1dee)
  | ==2428==ABORTING

Accepting input from '[STDIN]'
Usage for fuzzing: honggfuzz -P [flags] -- /mnt/scratch0/clusterfuzz/bot/builds/clusterfuzz-builds-honggfuzz_envoy_13d8ff3fd8b6e12ff5bbd32d951c40c9e1c6513f/revisions/ext_proc_unit_test_fuzz
[2023-05-22 14:48:58.839][2428][critical][assert] [./envoy/http/header_map.h:71] assert failure: valid().
AddressSanitizer:DEADLYSIGNAL
=================================================================
==2428==ERROR: AddressSanitizer: ABRT on unknown address 0x05390000097c (pc 0x7a908660600b bp 0x7ffdaf9e1c90 sp 0x7ffdaf9e19b0 T0)
SCARINESS: 10 (signal)
    #0 0x7a908660600b in raise /build/glibc-SzIz7B/glibc-2.31/sysdeps/unix/sysv/linux/raise.c:51:1
    #1 0x7a90865e5858 in abort /build/glibc-SzIz7B/glibc-2.31/stdlib/abort.c:79:7
    #2 0x1557b8f in Envoy::Http::LowerCaseString::LowerCaseString(std::__1::basic_string_view<char, std::__1::char_traits<char> >) [envoy/http/header_map.h:71](https://github.com/envoyproxy/envoy/blob/c2ae2211196a48b12d2e36d00c6c2889ae2f434a/http/header_map.h#L71):5
    #3 0x1685664 in Envoy::Extensions::HttpFilters::ExternalProcessing::MutationUtils::applyHeaderMutations(envoy::service::ext_proc::v3::HeaderMutation const&, Envoy::Http::HeaderMap&, bool, Envoy::Extensions::Filters::Common::MutationRules::Checker const&, Envoy::Stats::Counter&) /proc/self/cwd/source/extensions/filters/http/ext_proc/mutation_utils.cc:43:27
    #4 0x165dc14 in Envoy::Extensions::HttpFilters::ExternalProcessing::ProcessorState::handleHeadersResponse(envoy::service::ext_proc::v3::HeadersResponse const&) /proc/self/cwd/source/extensions/filters/http/ext_proc/processor_state.cc:87:31
    #5 0x163ae9e in Envoy::Extensions::HttpFilters::ExternalProcessing::Filter::onReceiveMessage(std::__1::unique_ptr<envoy::service::ext_proc::v3::ProcessingResponse, std::__1::default_delete<envoy::service::ext_proc::v3::ProcessingResponse> >&&) /proc/self/cwd/source/extensions/filters/http/ext_proc/ext_proc.cc:609:41
    #6 0x14c2fbf in operator() /proc/self/cwd/test/extensions/filters/http/ext_proc/unit_test_fuzz/ext_proc_unit_test_fuzz.cc:86:31
    #7 0x14c2fbf in __invoke<(lambda at test/extensions/filters/http/ext_proc/unit_test_fuzz/ext_proc_unit_test_fuzz.cc:82:28) &, envoy::service::ext_proc::v3::ProcessingRequest, bool> /usr/local/include/c++/v1/type_traits:3592:23
    #8 0x14c2fbf in __call<(lambda at test/extensions/filters/http/ext_proc/unit_test_fuzz/ext_proc_unit_test_fuzz.cc:82:28) &, envoy::service::ext_proc::v3::ProcessingRequest, bool> /usr/local/include/c++/v1/__functional/invoke.h:61:9
    #9 0x14c2fbf in operator() /usr/local/include/c++/v1/__functional/function.h:181:16
    #10 0x14c2fbf in std::__1::__function::__func<Envoy::Extensions::HttpFilters::ExtProc::UnitTestFuzz::TestOneProtoInput(envoy::extensions::filters::http::ext_proc::unit_test_fuzz::ExtProcUnitTestCase const&)::$_0::operator()(Envoy::Extensions::HttpFilters::ExternalProcessing::ExternalProcessorCallbacks&, envoy::config::core::v3::GrpcService const&, Envoy::StreamInfo::StreamInfo const&) const::'lambda'(envoy::service::ext_proc::v3::ProcessingRequest&&, bool), std::__1::allocator<Envoy::Extensions::HttpFilters::ExtProc::UnitTestFuzz::TestOneProtoInput(envoy::extensions::filters::http::ext_proc::unit_test_fuzz::ExtProcUnitTestCase const&)::$_0::operator()(Envoy::Extensions::HttpFilters::ExternalProcessing::ExternalProcessorCallbacks&, envoy::config::core::v3::GrpcService const&, Envoy::StreamInfo::StreamInfo const&) const::'lambda'(envoy::service::ext_proc::v3::ProcessingRequest&&, bool)>, void (envoy::service::ext_proc::v3::ProcessingRequest&&, bool)>::operator()(envoy::service::ext_proc::v3::ProcessingRequest&&, bool&&) /usr/local/include/c++/v1/__functional/function.h:355:12
    #11 0x1586861 in operator() /usr/local/include/c++/v1/__functional/function.h:508:16
    #12 0x1586861 in operator() /usr/local/include/c++/v1/__functional/function.h:1185:12
    #13 0x1586861 in ApplyImpl<const std::__1::function<void (envoy::service::ext_proc::v3::ProcessingRequest &&, bool)> &, std::__1::tuple<envoy::service::ext_proc::v3::ProcessingRequest &&, bool>, 0UL, 1UL> /proc/self/cwd/external/com_google_googletest/googlemock/include/gmock/internal/gmock-internal-utils.h:420:10
    #14 0x1586861 in Apply<const std::__1::function<void (envoy::service::ext_proc::v3::ProcessingRequest &&, bool)> &, std::__1::tuple<envoy::service::ext_proc::v3::ProcessingRequest &&, bool> > /proc/self/cwd/external/com_google_googletest/googlemock/include/gmock/internal/gmock-internal-utils.h:429:10
    #15 0x1586861 in Perform /proc/self/cwd/external/com_google_googletest/googlemock/include/gmock/gmock-actions.h:497:12
    #16 0x1586861 in PerformAction<void (envoy::service::ext_proc::v3::ProcessingRequest &&, bool)> /proc/self/cwd/external/com_google_googletest/googlemock/include/gmock/gmock-spec-builders.h:1441:12
    #17 0x1586861 in testing::internal::FunctionMocker<void (envoy::service::ext_proc::v3::ProcessingRequest&&, bool)>::UntypedPerformAction(void const*, void*) const /proc/self/cwd/external/com_google_googletest/googlemock/include/gmock/gmock-spec-builders.h:1556:12
    #18 0x50c8339 in testing::internal::UntypedFunctionMockerBase::UntypedInvokeWith(void*) /proc/self/cwd/external/com_google_googletest/googlemock/src/gmock-spec-builders.cc:452:24
    #19 0x1590021 in testing::internal::FunctionMocker<void (envoy::service::ext_proc::v3::ProcessingRequest&&, bool)>::Invoke(envoy::service::ext_proc::v3::ProcessingRequest&&, bool) /proc/self/cwd/external/com_google_googletest/googlemock/include/gmock/gmock-spec-builders.h:1593:15
    #20 0x1585db3 in Envoy::Extensions::HttpFilters::ExtProc::UnitTestFuzz::MockStream::send(envoy::service::ext_proc::v3::ProcessingRequest&&, bool) /proc/self/cwd/test/extensions/filters/http/ext_proc/unit_test_fuzz/mocks.h:20:3
    #21 0x162023f in Envoy::Extensions::HttpFilters::ExternalProcessing::Filter::onHeaders(Envoy::Extensions::HttpFilters::ExternalProcessing::ProcessorState&, Envoy::Http::RequestOrResponseHeaderMap&, bool) /proc/self/cwd/source/extensions/filters/http/ext_proc/ext_proc.cc:177:12
    #22 0x16225a6 in Envoy::Extensions::HttpFilters::ExternalProcessing::Filter::decodeHeaders(Envoy::Http::RequestHeaderMap&, bool) /proc/self/cwd/source/extensions/filters/http/ext_proc/ext_proc.cc:195:23
    #23 0x15c247b in Envoy::Http::FilterHeadersStatus Envoy::Extensions::HttpFilters::HttpFilterFuzzer::sendHeaders<Envoy::Http::StreamDecoderFilter>(Envoy::Http::StreamDecoderFilter*, test::fuzz::HttpData const&, bool) /proc/self/cwd/test/extensions/filters/http/common/fuzz/http_filter_fuzzer.h:145:46
    #24 0x14d648f in void Envoy::Extensions::HttpFilters::HttpFilterFuzzer::runData<Envoy::Http::StreamDecoderFilter>(Envoy::Http::StreamDecoderFilter*, test::fuzz::HttpData const&) /proc/self/cwd/test/extensions/filters/http/common/fuzz/http_filter_fuzzer.h:102:31
    #25 0x14bf07a in TestOneProtoInput /proc/self/cwd/test/extensions/filters/http/ext_proc/unit_test_fuzz/ext_proc_unit_test_fuzz.cc:93:10
    #26 0x14bf07a in LLVMFuzzerTestOneInput /proc/self/cwd/test/extensions/filters/http/ext_proc/unit_test_fuzz/ext_proc_unit_test_fuzz.cc:38:1
    #27 0x519eedb in main
    #28 0x7a90865e7082 in __libc_start_main /build/glibc-SzIz7B/glibc-2.31/csu/libc-start.c:308:16
    #29 0x140142d in _start
AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: ABRT (/lib/x86_64-linux-gnu/libc.so.6+0x4300b) (BuildId: 1878e6b475720c7c51969e69ab2d276fae6d1dee)
==2428==ABORTING

**Comment by yanjunxiang-google** [comment]
Test case:

config {
  grpc_service {
    envoy_grpc {
      cluster_name: "{"
    }
  }
  stat_prefix: "sr..&......e.p"
  mutation_rules {
    allow_envoy {
    }
    disallow_all {
    }
  }
}
request {
}
response {
  request_headers {
    response {
      header_mutation {
        remove_headers: "\0  002"
        remove_headers: "+d"
        remove_headers: "\184017"
        remove_headers: "+d"
      }
    }
  }
  dynamic_metadata {
  }
}

File: source/extensions/filters/http/ext_proc/mutation_utils.cc [hunk 1]
```diff
@@ -40,6 +40,11 @@ absl::Status MutationUtils::applyHeaderMutations(const HeaderMutation& mutation,
                                                  const Checker& checker,
                                                  Counter& rejected_mutations) {
   for (const auto& hdr : mutation.remove_headers()) {
+    if (!Envoy::Http::validHeaderString(hdr)) {
+      ENVOY_LOG(debug, "remove_headers contain null character, may not be removed.");
```
Comment (htuch) [inline source/extensions/filters/http/ext_proc/mutation_utils.cc:44]: Headers can be valid for reasons other than null characters, @yanavlasov

File: source/extensions/filters/http/ext_proc/mutation_utils.cc [hunk 2]
```diff
@@ -62,6 +67,12 @@ absl::Status MutationUtils::applyHeaderMutations(const HeaderMutation& mutation,
     if (!sh.has_header()) {
       continue;
     }
+    if (!Envoy::Http::validHeaderString(sh.header().key()) ||
+        !Envoy::Http::validHeaderString(sh.header().value())) {
+      ENVOY_LOG(debug, "set_headers contain null character in key or value, may not be appended.");
```
Comment (htuch) [inline source/extensions/filters/http/ext_proc/mutation_utils.cc:72]: Ditto

**Review by htuch — COMMENTED** [review]
Thanks, LGTM mostyl modulo comments. Needs a better description in commit message.

**Review by yanavlasov — CHANGES_REQUESTED** [review]
/wait

File: source/extensions/filters/http/ext_proc/mutation_utils.cc (hunk 1 above)
Comment (yanavlasov) [reply source/extensions/filters/http/ext_proc/mutation_utils.cc:44]: You can use the `HeaderUtility::headerNameIsValid` and `HeaderUtility::headerValueIsValid` for RFC compliant check.

**Review by yanavlasov — CHANGES_REQUESTED** [review]

File: source/extensions/filters/http/ext_proc/mutation_utils.cc (hunk 1 above)
Comment (yanjunxiang-google) [reply source/extensions/filters/http/ext_proc/mutation_utils.cc:44]: done

File: source/extensions/filters/http/ext_proc/ext_proc.cc [hunk 3]
```diff
@@ -765,7 +765,9 @@ void Filter::sendImmediateResponse(const ImmediateResponse& response) {
       const auto mut_status = MutationUtils::applyHeaderMutations(
           response.headers(), headers, false, immediateResponseChecker().checker(),
           stats_.rejected_header_mutations_);
-      ENVOY_BUG(mut_status.ok(), "Immediate response mutations should not fail");
+      if (!mut_status.ok()) {
+        ENVOY_LOG(error, "Immediate response mutations failed");
```
[the remark this answers is not in the document]
Comment (stevenzzzz) [reply source/extensions/filters/http/ext_proc/ext_proc.cc:769]: sorry, can we at least do LOG_EVERY_N here? I think this worth higher visibility. without explicitly creating an alerting rule on the stats, we can still capture it in error log.

[the remark this answers is not in the document]
Comment (yanjunxiang-google) [reply source/extensions/filters/http/ext_proc/ext_proc.cc]: OSS does not seem to have LOG_EVERY_N(). But we can change it back to error if we think having NULL characters  is a severe enough events which needs actions.  Also, I am open to change this into warn.

[the remark this answers is not in the document]
Comment (htuch) [reply source/extensions/filters/http/ext_proc/ext_proc.cc]: ENVOY_LOG_EVERY_NTH

File: test/extensions/filters/http/ext_proc/mutation_utils_test.cc [hunk 4]
```diff
@@ -188,6 +188,46 @@ TEST(MutationUtils, TestNonAppendableHeaders) {
...
+  s = mutation.add_set_headers();
+  // Test header value contains null character.
+  s->mutable_header()->set_key("x-append-this");
+  s->mutable_header()->set_value("value\r");
+  EXPECT_CALL(rejections, inc());
+  EXPECT_FALSE(
+      MutationUtils::applyHeaderMutations(mutation, headers, false, checker, rejections).ok());
+}
+
+TEST(MutationUtils, TestRemoveHeaderWithNullCharacter) {
```
Comment (htuch) [inline test/extensions/filters/http/ext_proc/mutation_utils_test.cc:217]: Nit: this test and above are using invalid characters, not null. It should be `TestRemoveHeadersWithInvalidChar` etc.

File: test/extensions/filters/http/ext_proc/ext_proc_integration_test.cc [hunk 5]
```diff
@@ -1079,6 +1079,21 @@ TEST_P(ExtProcIntegrationTest, GetAndRespondImmediately) {
   EXPECT_EQ("{\"reason\": \"Not authorized\"}", response->body());
 }
 
+TEST_P(ExtProcIntegrationTest, GetAndRespondImmediatelyWithNullCharacter) {
```
Comment (yanavlasov) [inline test/extensions/filters/http/ext_proc/ext_proc_integration_test.cc:1082]: The test name implies null character, but it is actually '\n'

**Review by yanavlasov — CHANGES_REQUESTED** [review]
/wait

File: test/extensions/filters/http/ext_proc/ext_proc_integration_test.cc (hunk 5 above)
Comment (yanjunxiang-google) [reply test/extensions/filters/http/ext_proc/ext_proc_integration_test.cc]: done

File: source/extensions/filters/http/ext_proc/ext_proc.cc [hunk 6]
```diff
@@ -765,7 +765,9 @@ void Filter::sendImmediateResponse(const ImmediateResponse& response) {
       const auto mut_status = MutationUtils::applyHeaderMutations(
           response.headers(), headers, false, immediateResponseChecker().checker(),
           stats_.rejected_header_mutations_);
-      ENVOY_BUG(mut_status.ok(), "Immediate response mutations should not fail");
+      if (!mut_status.ok()) {
+        ENVOY_LOG_EVERY_POW_2(error, "Immediate response mutations failed");
```
Comment (htuch) [inline source/extensions/filters/http/ext_proc/ext_proc.cc:769]: Nit: include the `mut_status` error text in the log as well.

**Review by htuch — COMMENTED** [review]
LGTM modulo nit.

File: source/extensions/filters/http/ext_proc/ext_proc.cc (hunk 6 above)
Comment (yanjunxiang-google) [reply source/extensions/filters/http/ext_proc/ext_proc.cc]: done

**Review by yanavlasov — APPROVED** [review]

**Pull request merged**